challenges/pt-br/new-backend-challenge/README.md

```diff
@@ -0,0 +1,92 @@
+# Back-end challenge
+
+Olá!
+
+Estamos felizes por você ter chegado até aqui! Agora é a hora de conhecermos um pouco mais da sua bagagem técnica. Esperamos que goste do Desafio!
+
+## API de carrinho
+
+Você deverá implementar uma API HTTP (JSON) de e-commerce (venda online) e que terá um endpoint de carrinho (checkout). Esse endpoint aceitará uma requisição com método POST, header X-USER-ID contendo o id de um usuário e a estrutura do payload de requisição deve seguir o exemplo:
+
+```javascript
+{
+    "products": [
+        {
+            "id": 1,
+            "quantity": 1 // Quantidade a ser comprada do produto
+        }
+    ]
+}
+```
+
+Em resumo a requisição deve conter uma lista de produtos e a quantidade de cada um a ser comprado.
+
+## Regras
+
+Após receber a requisição deverão ser aplicadas as seguintes regras:
+
+### Regra número 1
+
+Para cada produto você precisará calcular a porcentagem de desconto e isso deve ser feito consumindo um serviço gRPC fornecido por nós para auxiliar no seu teste. Utilize a [imagem Docker](https://hub.docker.com/r/hashorg/hash-mock-discount-service) para subir esse serviço de desconto e o [arquivo proto](https://github.com/hashlab/hiring/blob/master/challenges/pt-br/new-backend-challenge/discount.proto) para gerar o cliente na linguagem escolhida. Você pode encontrar como gerar um cliente gRPC nas documentações oficiais da ferramenta e em outros guias encontrados na internet.
+
+### Regra número 2
+
+Caso o serviço de desconto esteja indisponível o endpoint deverá continuar funcionando porém não vai realizar o cálculo com desconto.
+
+### Regra número 3
+
+Durante o cálculo do valor total deverá ser considerada a quantidade do produto.
+
+### Regra número 4
+
+Deverá ser verificado se é o aniversário do usuário recebido via header e caso seja você deve adicionar um produto brinde ao carrinho.
```

Comment:
Poderíamos enriquecer essa regra dizendo por exemplo que o produto de menor valor deverá ser considerado o brinde.

Reply:
@paulovitin, se o produto de menor valor ainda for um valor que o usuário vai precisar pagar o produto ainda não é um brinde.

Aprofundando um pouco mais em e-commerce, existe uma diferença entre produto brinde e um produto que foi dado um % de desconto, então acho que seria legal pensar melhor na semântica da palavra **brinde**.

Aqui na minha visão não é um brinde mas sim um desconto aplicado no produto apenas.

Reply:
@deepakvashist @kpilon @paulovitin o que eu tinha entendido, era que um produto adicionado no carrinho na forma de brinde, a diferença é que vai ter uma flag "is_gift=true" e seu valor não será calculado no total do carrinho e nesse caso, pra simplificar, não importa muito pra gente se é um produto que já está no carrinho ou não. Apenas será adicionado outra entrada dele, como gift=true, caso já exista no carrinho.

Mas eu acho que se essa regra estiver ok mesmo, vale a gente dar uma melhorada na regra do brinde.

Reply:
Vale deixar mais claro que pode ser um produto que ele ja tenha no carrinho e complementar que deveria ou nao ser o produto de menor valor